### ISSUE TITLE
Cannot connect to Car Scanner ELM OBD2？

### ISSUE THREAD

**Issue by Okita-Sawa** [HerrRiebmann/KDS2Bluetooth#9]
I connected to Car Scanner ELM OBD2, but it disconnected for a few minutes and then wouldn't connect again.I re-uploaded the firmware, replaced the new Arduino and the new l9637d, but it still can't connect to the Car Scanner ELM OBD2.
![QQ图片20241128214225](https://github.com/user-attachments/assets/d87d99e0-b7dc-47c6-acf8-e5b65171f22e)
This is a screenshot of XT Commands, it seems it can communicate with the ecu?

**Comment by HerrRiebmann** [comment]
Hi,
I tested the solution with this exact Application and it worked fine.
But what I see is that you have some things quite strange:
Throttle and SubThrottle seem to be strange.
RAW-Mode is active, which removes the (in this case) Kawasaki specific conversions and just channels the commands from the Car Scanner Application.

So I´d set the Kawasaki Protocol again, by `XTP01` to have Throttle and stuff straight.
Then deactivate the Raw mode by `XTR0`
`XTE `shows the last errors, which might help. But due to the already wrong setup, I´d clear the error storage first by `XTER`

Strange that "Storage" is set to true as well. This should only be true when another arduino is connected, to write anything onto an SD-Card. Or if it was set manually. This would require a unique solution, which I used for debugging and testing purposes.

See all the other XT commands for setting up the device here:
https://github.com/HerrRiebmann/KDS2Bluetooth/tree/master/Bike_Diagnostic_System#xt-commands---custom-setupfunctions

Which Bike do you have?

**Comment by Okita-Sawa** [comment]
Thanks!
I restored the connection using XT Commands.It's working properly now.
I found the problem. There is something wrong with the plug that connects to the vehicle. The vibration will cause the k-line to be disconnected when driving on the road.If you do not restart the Arduino at this time and click on the connection of the APP, it will enter "connecting to ecu using protocol" and it will not recognize the protocol.Then restart Arduino and enter XT Commands, RAW-MODE will become TRUE.After executing the XT Commands you provided, the APP can successfully connect again.
My bike is 2017 kawasaki z250.

**Comment by HerrRiebmann** [comment]
Ahh, allright! That habit sounds strange, but understandable :)
I think I test a lot, but not disconnecting the ECU with the App still running. At least not this one.
My camera just goes on requesting and I got about 3-4 seconds no data.

But I might test this tomorrow with my emulator. Maybe we can find an easy fix to not mess up all the settings.

**Comment by HerrRiebmann** [comment]
And if you like to, please run a test procedure and give me the results?
This would be amazing! I got lots of examples but still try to figure out some magic 👯 

**XTT1** -> Set Debug Mode to output everything
**XTZ** -> TestMode (ECU Id, & All PID´s)

If possible, run XTZ once with the engine off and again with the engine turned on.
The Serial Bluetooth Terminal has a function to log to a file. This will be needed to catch all data.
I might give you some more insights about you bike then.

**Comment by Okita-Sawa** [comment]
Of course, but I'm on vacation now and will get back to you in a few days.:)
